staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -139,7 +142,23 @@ func newStore(c *clientv3.Client, codec runtime.Codec, newFunc, newListFunc func
 	w.getCurrentStorageRV = func(ctx context.Context) (uint64, error) {
 		return storage.GetCurrentResourceVersionFromStorage(ctx, s, newListFunc, resourcePrefix, w.objectType)
 	}
-	return s
+	if utilfeature.DefaultFeatureGate.Enabled(features.ConsistentListFromCache) {
+		ctx, cancel := context.WithCancel(context.Background())
+		// Cancel context after 10 minutes.
+		go func() {
+			select {
+			case <-time.After(10 * time.Minute):
+				cancel()
+			case <-ctx.Done():
+			}
+		}()
```

Comment:
```suggestion
		ctx, cancel := context.WithTimeout(context.Background(), 10*time.Minute)
```

Reply:
Cancel function should be put on the Storage to allow cancelling during shutdown before 10 minute passes.

Reply:
Still I'm not sure we should give a timeout here, we just need to make sure that shutdown of apiserver calls shutdown of storage, which cancels this.

Reply:
The intention was when etcd being ready after api server. please see https://github.com/kubernetes/kubernetes/pull/124612#discussion_r1590921554
@p0lyn0mial